src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -219,7 +220,11 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                 {
                     var result = await ptr(this, context.IncrementStackDepthCounter(childContext), node.IRContext, args);
 
-                    Contract.Assert(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType);
+                    if (IfFunction.CanCheckIfReturn(func))
```

Comment:
>CanCheckIfReturn [](http://example.com/codeflow?start=35&length=16)

Note to future readers - this is needed to loosen the assert.
We can't fully determine when the type doesn't matter, so we just ignore If alttogether.

 If(false, "abc1", {x:3}) // Behavior

Reply:
Check if node.IRContext.ResultType is DType.Unknown